api/circulation.py

```diff
@@ -993,15 +993,17 @@ def fulfill(self, patron, pin, licensepool, delivery_mechanism, part=None, fulfi
                     part=part, fulfill_part_url=fulfill_part_url
                 )
 
-                if loan and fulfillment.can_cache_manifest:
-                    loan.cached_manifest = fulfillment.content.encode("raw_unicode_escape")
-                    loan.cached_content_type = fulfillment.content_type
-
             if not fulfillment or not (
                 fulfillment.content_link or fulfillment.content
             ):
                 raise NoAcceptableFormat()
 
+            # Now that the additional API calls have been made,
+            # fulfillment data can be saved to the loan for future retrieval
+            if loan and fulfillment.can_cache_manifest:
+                loan.cached_manifest = fulfillment.content.encode("raw_unicode_escape")
+                loan.cached_content_type = fulfillment.content_type
+
```

Comment:
Did this fix the problem you were seeing? I don't understand why it would have.

Reply:
Yes, I think it's because `fulfillment.can_cache_manifest` isn't set to `True` until `fulfillment.content` is accessed, which kicks off `Axis360FulfillmentInfo.do_fetch()` that sets `fulfillment.can_cache_manifest=True`.

If this block of code fits better where it was then I think something like `if loan and fulfillment.content and fulfillment.can_cache_manifest:` would work too.

Reply:
No, your explanation makes sense.